### PR TITLE
#3812 Paper onChange action with Pipeline API

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3812 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+
+import * as context from 'context';
+
+const pipeline = () => {
+  const { updatePipeline } = context.usePipelineActions();
+  const { currentPipeline } = context.useWorkspace();
+
+  const update = async params => {
+    await updatePipeline({ ...params });
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: Is this necessary?

File: ohara-manager/client/src/components/Pipeline/Pipeline.js [hunk 2]
```diff
@@ -195,6 +197,7 @@ const Pipeline = () => {
                   ref={paperRef}
                   onCellSelect={element => setSelectedCell(element)}
                   onCellDeselect={() => setSelectedCell(null)}
+                  onChange={_.debounce(paperApi => updateCells(paperApi), 1000)}
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Pipeline.js]: _.debounce(func, [wait=0], [options={}]) 預設行為是 wait 後執行 func。假如需要在 debounce 的開始點及結束點執行 func，可以加上 options = { 'leading': true, 'trailing': true }

Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Pipeline.js]: 目前onChange的事件都是比較不重要的更新，所以一段時間後再儲存就可以了。

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: yes other fun need

Comment (eechih) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: 看起來似乎只有把 updatePipeline 改名為 update，是不是直接寫 const { updatePipeline: update } = context.usePipelineActions() 比較簡單

Comment (eechih) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: updatePipeline() 已經是 async function

Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: Sorry, didn't make this clear 😅 I mean shouldn't just pass down the params like:
```js
const update = async params => {
    await updatePipeline(params);
```

Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: But @eechih is also right, this wrapping seems not necessary

File: ohara-manager/client/src/components/Pipeline/Pipeline.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/Pipeline/Pipeline.js]: OK

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/pipeline.js]: ok~~thanks.

**Comment by wu87988622** [comment]
retry manager

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by wu87988622** [comment]
retry manager

**Comment by wu87988622** [comment]
retry manager

**Pull request merged**